exercises/palindrome-products/palindrome_products_test.go

```diff
@@ -37,10 +38,14 @@ var testData = []struct {
 		""},
 	{4, 10, Product{}, Product{}, "No palindromes"},
 	{10, 4, Product{}, Product{}, "fmin > fmax"},
-	/* bonus curiosities.  (can a negative number be a palindrome?
-	// most say no
-	{-99, -10, Product{}, Product{}, "Negative limits"},
-	// but you can still get non-negative products from negative factors.
+
+	/* Bonus curiosities.  Can a negative number be a palindrome? Most say no
+	{-99, -10,
+		Product{},
+		Product{},
+	"Negative limits"},
```

Comment:
doh, that's unfortunate. Another good reason why we'd want to uncomment it, I guess, we wouldn't have been able to catch that.

My first suggestion is... is it possible to convert that value back to an int? `int(^uint(0) >> 1)` or something? Since it's right-shifted by one, that means that should become the largest positive signed int value, which I suppose is the intent of the test (to make it overflow).

I looked to see if there were any constants that can be used, but there's `MaxInt32` and `MaxInt64` and we would need to know the size of `int` before using either of those. I guess that's why `^uint(0) >> 1` is being used. Anyway, from http://stackoverflow.com/questions/6878590/the-maximum-value-for-an-int-type-in-go it looks like `int(^uint(0) >> 1)` is indeed the way to go.